File: frontend/src/types/Issue.ts
import { Book } from './Book';
import { User } from './User';

export interface Issue {
    id: number;
    book: Book;
    bookId: number;
    bookTitle: string;
    bookIsbn: string;
    user: User;
    userId: number;
    userName: string;
    memberName: string;
    memberEmail: string;
    issueDate: string;
    dueDate: string;
    returnDate?: string;
    fine: number;
    status: 'ISSUED' | 'RETURNED' | 'OVERDUE';
    remarks?: string;
    createdAt: string;
    updatedAt: string;
}

export interface IssueFormData {
    bookId: number;
    memberName: string;
    memberEmail: string;
    remarks?: string;
}

File: backend/src/main/java/com/library/management/model/Issue.java
package com.library.management.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "issue", indexes = {
        @Index(name = "idx_book_id", columnList = "book_id"),
        @Index(name = "idx_status", columnList = "status")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "book_id", nullable = false)
    private Book book;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotBlank
    @Column(nullable = false)
    private String memberName;

    @NotBlank
    @Email
    @Column(nullable = false)
    private String memberEmail;

    @NotNull
    @Column(nullable = false)
    private LocalDate issueDate;

    @NotNull
    @Column(nullable = false)
    private LocalDate dueDate;

    private LocalDate returnDate;

    @Column(precision = 10, scale = 2)
    private BigDecimal fine = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueStatus status = IssueStatus.ISSUED;

    @Column(columnDefinition = "TEXT")
    private String remarks;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    public enum IssueStatus {
        ISSUED, RETURNED, OVERDUE
    }
}
